pages/blog/_posts/2020-10-01-what-is-short-messages.md

```diff
@@ -0,0 +1,214 @@
+layers between *SCTP* and *MAP* are *M3UA* (Message Transfer Part 3 User
+Adaptation Layer) and *SCCP* (Signalling Connection Control Part) which
+both handles handshaking, routing, and resilience.
+
+The *MAP* layer is used when talking to some of the telco nodes such
+as *HLR* (Home location registry), *VLR* (Visitor location registry),
+*MSC* (Mobile switching center), *SGSN* (Serving *GPRS* [ackronym in
+ackronyms; go telco!] support node) and the *SMSC* (Short message
+service center).
+
+# MAP versions and TCAP dialogues
+
+There are three iterations of *MAP*, v1, v2, and v3, and all messages
+almost always comes in pair, an acknowledgement for each sent message
+(so called `Invoke` and `ReturnResult`).
+
+To determine which version to use between two nodes, the sending node
+tries to start the transaction (called a dialogue) by sending a *TCAP*
+`Begin` message with the *MAP* message and it's highest compatible
+version. If the receiving node cannot talk that version, it sends a
+*TCAP* `Abort` message with it's highest compatible version. For v1 there
+might not even be a reason, but the sending node might try to send the
+*MAP* message as v1 anyway.
+
+In my head it goes like this:
+
+```
+Node 1: "Hi, I want to talk to you about this in the latest version"
+Node 2: "No I don't understand you, but we can talk about version 2 instead"
+Node 1: "Ok, then I want to talk to you about this in version 2 instead"
+Node 2: "Aah, now I see..."
+```
+
+Or maybe
+
+```
+Node 1: "Hi, I want to talk to you about this in the latest version"
+Node 2: "No"
+Node 1: "Ok, then I want to talk to you about this in version 1 instead"
+Node 2: "Maybe I will talk to you, maybe I will not"
+```
+
+For *TCAP* dialogues there are (mainly) four message types.  `Begin`,
+`Continue`, `End`, `Abort`. Each of the types have an ID (or two, as I
+said, telco is complicated), a component and a dialogue part. The
+component contains the *MAP* messages, and the dialogue part contains
+the version negotiation, but it is only used in the first message from
+both nodes for the version negotiation.
+
+_I think this covers most of it, let's get back to the fun part._
+
+# How does SMS work?
+
+*SMS* was initally implemented because of the wish to send text
+messages to pagers using the phoneline when it was not in use for
+phonecalls. It was decided at a meeting in Oslo to be released to the
+public when some French and German company understood it's
+value. (Don't quote me on any of this).
+
+When you send an *SMS*, a signal goes towards the *SMSC* node in your
+current *SS7* network. This signal is in a form of a packet, and is
+called `MO-Forward-SM`. It stands for "Mobile Originating Forward Short
+Message" meaning it started from your (mobile-)phone.
+
+The *SMSC* then takes your *SMS* and tries to find out where to send
+it. For instance if the recipient's phone is on the same
+operator/network or if it should route it to another operators
```

Comment:
It actually does not really think about "other" smsc. It just asks where to send the message, and that other entity will be assumed to be the recipient's device. Whether it is or not. It's just that the other side may choose to send it to it's SMSC instead of the device, which is there the `SRI-for-SM` shenanigans come in...